src/index.js

```diff
@@ -75,4 +102,28 @@ export default class SaucelabsConnector {
 
         await closeTunnel();
     }
+
+    async waitForFreeMachines (machinesCount, requestInterval, attemptsCount) {
+        var attempt          = 1;
+        var isMachinesEnough = false;
+
+        while (!isMachinesEnough && attempt <= attemptsCount) {
+            var freeMachinesCount = await this._getFreeMachinesCount();
+
+            isMachinesEnough = await freeMachinesCount >= machinesCount;
+
+            if (!isMachinesEnough) {
+                process.stdout.write('The amount of free machines is below than requested.\n');
+                attempt++;
+
+                if (attempt <= attemptsCount)
+                    await wait(requestInterval);
+            }
+            else
+                process.stdout.write('The amount of free machines is enough for request.\n');
+        }
+
+        if (!isMachinesEnough)
+            throw new Error('There are no free machines');
+    }

```

Comment:
There are some duplicated checks (for example `if (attempt <= attemptsCount)`).

Let's write in this way:
```js
var attempts = 0;

while (attempts < maxAttemptsCount) {
    var freeMachinesCount = await this._getFreeMachinesCount();
    
    if(freeMachinesCount >= machinesCount)
        return;

    //log here about available machines

    await wait(requestInterval);
    attemps++;
}

throw new Error('There are no free machines');
```